src/System.IO/tests/TextWriter/TextWriterTests.cs

```diff
@@ -0,0 +1,598 @@
+            }
+        }
+
+        [Fact]
+        public void WriteLineBoolTest()
+        {
+            using (TextWriterImpl tw = GetTextWriter())
+            {
+                tw.WriteLine(true);
+                Assert.Equal("True" + tw.NewLine, tw.Text);
+
+                tw.Clear();
+                tw.WriteLine(false);
+                Assert.Equal("False" + tw.NewLine, tw.Text);
+            }
+        }
+
+        [Fact]
+        public void WriteLineIntTest()
+        {
+            using (TextWriterImpl tw = GetTextWriter())
+            {
+                tw.WriteLine(int.MinValue);
+                Assert.Equal(int.MinValue.ToString() + tw.NewLine, tw.Text);
+
+                tw.Clear();
+                tw.WriteLine(int.MaxValue);
+                Assert.Equal(int.MaxValue.ToString() + tw.NewLine, tw.Text);
+            }
+        }
+
+        [Fact]
+        public void WriteLineUIntTest()
+        {
+            using (TextWriterImpl tw = GetTextWriter())
+            {
+                tw.WriteLine(uint.MinValue);
+                Assert.Equal(uint.MinValue.ToString() + tw.NewLine, tw.Text);
+
+                tw.Clear();
+                tw.WriteLine(uint.MaxValue);
+                Assert.Equal(uint.MaxValue.ToString() + tw.NewLine, tw.Text);
+            }
+        }
+
+        [Fact]
+        public void WriteLineLongTest()
+        {
+            using (TextWriterImpl tw = GetTextWriter())
+            {
+                tw.WriteLine(long.MinValue);
+                Assert.Equal(long.MinValue.ToString() + tw.NewLine, tw.Text);
+
+                tw.Clear();
+                tw.WriteLine(long.MaxValue);
+                Assert.Equal(long.MaxValue.ToString() + tw.NewLine, tw.Text);
+            }
+        }
+
+        [Fact]
+        public void WriteLineULongTest()
+        {
+            using (TextWriterImpl tw = GetTextWriter())
+            {
+                tw.WriteLine(ulong.MinValue);
+                Assert.Equal(ulong.MinValue.ToString() + tw.NewLine, tw.Text);
+
+                tw.Clear();
+                tw.WriteLine(ulong.MaxValue);
+                Assert.Equal(ulong.MaxValue.ToString() + tw.NewLine, tw.Text);
+            }
+        }
+
+        [Fact]
+        public void WriteLineFloatTest()
+        {
+            using (TextWriterImpl tw = GetTextWriter())
+            {
+                tw.WriteLine(float.MinValue);
+                Assert.Equal(float.MinValue.ToString() + tw.NewLine, tw.Text);
+
+                tw.Clear();
+                tw.WriteLine(float.MaxValue);
+                Assert.Equal(float.MaxValue.ToString() + tw.NewLine, tw.Text);
+
+                tw.Clear();
+                tw.WriteLine(float.NaN);
+                Assert.Equal(float.NaN.ToString() + tw.NewLine, tw.Text);
+            }
+        }
+
+        [Fact]
+        public void WriteLineDoubleTest()
+        {
+            using (TextWriterImpl tw = GetTextWriter())
```

Comment:
You could have a utility function like
`void AssertWriteLine(object value, string expected)`
and call it like
`AssertWriteLine(double.MinValue, double.MinValue.ToString());`
and it would take care of creating the text writer, writing, clearing, and the newline. Making these tests 3 lines each

Reply:
Not sure... but will it end up invoking the TextWriter::WriteLine(object) overload instead of exercising the individual overloads?

Reply:
Ah - you're quite right. Disregard.